### ISSUE TITLE
feature request: application protection

### ISSUE THREAD

**Issue by macarthor** [MayGo/tockler#246]
would you please add some features like application protection/lock, as means only authorized people can open the application and read the data, for any reasons

**Comment by MayGo** [comment]
Yep, I can see uses for some kind of lock. (password)
Do you need only the main window locked or the tray window also?

**Comment by macarthor** [comment]
> Yep, I can see uses for some kind of lock. (password) Do you need only the main window locked or the tray window also?

hi MayGo,

all UIs/data have two aspects: read and write. IMO read should be ok for all users, while write (maybe clear data for this specific time tracker application) should be protected.

of course the protection is within the application scope. access to the file system (delete or edit the DB file) is beyond the application scope.

i have not used all parts of your great application, and cannot give suggestions for specific feature if it should be protected or not. what i can give is a general suggestion :)

btw, why i have this idea of the feature request. my kid is taking online lessions at home all day due to C-19, and i wanna track his usage of the PC, and keep him from clearing the data so that i can get the full result...

**Comment by macarthor** [comment]
another scenario to protect is to close the application process completely when unauthorized user would stop the tracking.

**Comment by MayGo** [comment]
I agree that this feature would be best to make sure the kid is doing homework, not playing some games (eg: browser-based). For macOS we have ScreenTime, but that gives a very vague understanding of application usage.
I had a similar problem at one point when classes were remote, but then the kid did not know too much computer stuff.
